src/UserHandler.cpp

```diff
@@ -192,11 +198,18 @@ void UserHandler::logout(std::string& line) {
     if (ansCmd == ERROR) {
         std::cout << ans.getBody() << std::endl;
     }
+    else {// ansCmd == RECEIPT
+        _inventory.clear();
+        while(!_queues.framesFromServer.empty()) _queues.framesFromServer.pop();
```

Comment:
:laughing: 
I asked myself if it needed and decided to pass. 
**sync on the queue is required...**

Reply:
Good idea, will add!

Reply:
I think you should use lock_guard. I'm not perfectly sure whats the different, but i rather not wait to find out